### PR TITLE
Use matplotlib more transparently

### PR DESCRIPTION
**Pull request by vgalisson**
As I suggested in the [PR](https://github.com/vgalisson/pySankey/pull/1) in my own repo :

> Matplotlib gives a lot of flexibility in terms of plot sizes, aspect ratios, adding annotations, styling, and so on: all of which are valuable to people creating sankey diagrams. Currently, the sankey function uses a number of extra keyword arguments to ape a tiny portion of that flexibility, making it harder to read and maintain, while taking a lot of valuable power away from the user.

> sankey could, instead, take a matplotlib Axes object, which the user can customize to their liking, and where the user also has access to the Figure to customize. Then they can choose what to do with it - showing it, saving it with much more flexibility.

> That's what I tried to implement here by removing some parameters and returning a matplotlib Axes.

> Additionally, I reorganized and added some tests.

It softly makes changes without breaking existing usage and adds deprecation warnings.

### CODE REVIEW

File: pysankey/sankey.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # -*- coding: utf-8 -*-
-"""
+r"""
 Produces simple Sankey Diagrams with matplotlib.
 @author: Anneya Golob & marcomanz & pierre-sassoulas & jorwoods
```
Comment (Pierre-Sassoulas) [inline pysankey/sankey.py]: @vgalisson you should add your name here :)

File: pysankey/tests/generic_test.py [hunk 2]
```diff
@@ -3,14 +3,55 @@
...
+        cls.colorDict = ''
+
     def tearDown(self):
         path = "{}.png".format(self.figure_name)
         # Comment this to check the resulting image
         if os.path.exists(path):
             os.remove(path)
+
+
+class TestFruit(GenericTest):
```
Comment (Pierre-Sassoulas) [inline pysankey/tests/generic_test.py]: Nice job upping the coverage to almost 100 %!

Comment (vgalisson) [reply pysankey/tests/generic_test.py]: Thanks :)

File: pysankey/sankey.py (hunk 1 above)
Comment (vgalisson) [reply pysankey/sankey.py]: done

**Comment by Pierre-Sassoulas** [comment]
I released a 1.1.0 version of PySankeyBeta.

**Comment by Pierre-Sassoulas** [comment]
Thanks you for this great pull request, much appreciated.

**Pull request merged**